### PR TITLE
Refactor Tile Inspector widget definitions

### PR DESCRIPTION
**Pull request by Broxzier**
This is currently ~~a Work In Progress~~ ready for review, as part of #12329.

Tasks:
 - [x] 'Main' widgets (top buttons, spinners, list, groupboxes, etc)
 - [x] Surface-specific widgets
 - [x] Path-specific widgets
 - [x] Track-specific widgets
 - [x] Scenery-specific widgets
 - [x] Entrance-specific widgets
 - [x] Wall-specific widgets
 - [x] Large scenery-specific widgets
 - [x] Banner-specific widgets
 - [x] Corrupt element-specific widgets
 - [x] ~~Better name constants like `SUR_GBPB`~~ They're now removed
 - [x] Introduce function for positioning the widgets
 - [x] Fix horizontal scrollbar

This PR also corrects some incorrect pixel offsets, such as these:

### CODE REVIEW

**Comment by AaronVanGeffen** [comment]
@Broxzier Would you like some help with this?

File: src/openrct2-ui/windows/TileInspector.cpp [hunk 1]
```diff
@@ -353,13 +353,13 @@ static rct_widget WallWidgets[] = {
...
-constexpr int32_t LAR_GBDB = LAR_GBPT + GROUPBOX_PADDING;  // Large scenery group box info bottom
-constexpr int32_t LAR_GBDT = LAR_GBDB + 20 + 3 * 11;       // Large scenery group box info top
+constexpr int32_t NumLargeSceneryProperties = 1;
+constexpr int32_t NumLargeSceneryDetails = 3;
+constexpr int32_t LargeSceneryPropertiesHeight = 16 + NumLargeSceneryProperties * 21;
+constexpr int32_t LargeSceneryDetailsHeight = 20 + NumLargeSceneryDetails * 11;
 static rct_widget LargeSceneryWidgets[] = {
     MAIN_TILE_INSPECTOR_WIDGETS,
-      SPINNER_WIDGETS      (1,  GBBL(1), GBBR(1), GBBT(WH - LAR_GBPT, 0) + 3, GBBB(WH - LAR_GBPT, 0) - 3,   STR_NONE,  STR_NONE),  // WIDX_LARGE_SCENERY_SPINNER_HEIGHT{,_INCREASE,_DECREASE}
+    MakeSpinnerWidgets(PropertyRowCol({ 12, 0 }, 0, 1), PropertyButtonSize, WindowWidgetType::Spinner, WindowColour::Secondary), // WIDX_BANNER_SPINNER_HEIGHT{,_INCREASE,_DECREASE}
```
Comment (Broxzier) [inline src/openrct2-ui/windows/TileInspector.cpp:362]: Incorrect WIDX

File: src/openrct2-ui/windows/TileInspector.cpp [hunk 2]
```diff
@@ -396,14 +403,28 @@ static rct_widget *PageWidgets[] = {
...
+    TileInspectorGroupboxSettings settings{};
+    decltype(settings.properties_bottom_offset) offsetSum = 0;
+    settings.properties_bottom_offset = (offsetSum += PADDING_BOTTOM);
+    settings.properties_top_offset = (offsetSum += propertiesHeight);
+    settings.details_bottom_offset = (offsetSum += GROUPBOX_PADDING);
+    settings.details_top_offset = (offsetSum += detailsHeight);
+    return settings;
+}
+
+constexpr TileInspectorGroupboxSettings PageGroupBoxSettings[] = {
```
Comment (Broxzier) [inline src/openrct2-ui/windows/TileInspector.cpp:427]: static

**Comment by Broxzier** [comment]
This PR is now ready for review.

File: src/openrct2-ui/windows/TileInspector.cpp [hunk 3]
```diff
@@ -180,199 +180,198 @@ constexpr int32_t MIN_WH = 130;
...
-    { WindowWidgetType::Checkbox,         1,  CHK(GBBL(1) + 14 * 0, GBBT(WH - SUR_GBPT, 2) + 7 * 1),  STR_NONE,   STR_NONE }, // WIDX_SURFACE_CHECK_CORNER_W
-    { WindowWidgetType::Checkbox,         1,  GBBF(WH - SUR_GBPT, 0, 4),  STR_TILE_INSPECTOR_SURFACE_DIAGONAL,    STR_NONE }, // WIDX_SURFACE_CHECK_DIAGONAL
+    MakeSpinnerWidgets(PropertyRowCol({ 12, 0 }, 0, 1), PropertyButtonSize, WindowWidgetType::Spinner, WindowColour::Secondary), // WIDX_SURFACE_SPINNER_HEIGHT{,_INCREASE,_DECREASE}
+    MakeWidget(PropertyRowCol({ 12, 0 }, 1, 0),         PropertyButtonSize, WindowWidgetType::Button,  WindowColour::Secondary, STR_TILE_INSPECTOR_SURFACE_REMOVE_FENCES), // WIDX_SURFACE_BUTTON_REMOVE_FENCES
+    MakeWidget(PropertyRowCol({ 12, 0 }, 1, 1),         PropertyButtonSize, WindowWidgetType::Button,  WindowColour::Secondary, STR_TILE_INSPECTOR_SURFACE_RESTORE_FENCES), // WIDX_SURFACE_BUTTON_RESTORE_FENCES
+    MakeWidget(PropertyRowCol({ 12, 0 }, 3, 1) + ScreenCoordsXY{ 14 * 1, 7 * 0}, { 12, 12 }, WindowWidgetType::Checkbox, WindowColour::Secondary), // WIDX_SURFACE_CHECK_CORNER_N
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/TileInspector.cpp:258]: This 14 obviously has meaning. I guess it’s the row height? Can you create a constant called RowHeight and use it?

Comment (Broxzier) [reply src/openrct2-ui/windows/TileInspector.cpp]: 14 and 7 are horizontal and vertical offsets respectively for the checkboxes groups, the multiplication shows by how much. I've added a helper function for them instead.

**Review by Gymnasiast — COMMENTED** [review]
Code looks reasonable, would like to test.

**Comment by AaronVanGeffen** [comment]
Fwiw, the changelog entry hasn't been moved to v0.3.5+ yet.

**Comment by Broxzier** [comment]
Good catch.

**Comment by Gymnasiast** [comment]
Fixed in https://github.com/OpenRCT2/OpenRCT2/commit/b78908caa20310452340769a9ec050ae70756e66.

**Pull request merged**